### PR TITLE
adding logrotate support

### PR DESCRIPTION
**Pull request by prometheanfire**
fixes #45

### CODE REVIEW

**Comment by mattjbarlow** [comment]
I think you should have a boolean logrotate attribute for setting logrotate on or off. I'd be fine with setting it to `true` by default, but there should be a way to turn logrotate off if desired. So, your `only_if` statement [here](https://github.com/rackspace-cookbooks/varnish/blob/pfire-logrotate/libraries/varnish_log.rb#L117) would also check for the truthiness of the new_resource.logrotate attribute. Let me know what you think.

**Comment by jujugrrr** [comment]
Hi,
I think if we want to setup logrotate differently that what's been provided by the package, we should do it while we integrate the varnish cookbook, but not inside the varnish cookbook/resource itself. It sounds out of scope or me. If really we want to manage it form the cookbook, could we use https://github.com/stevendanna/logrotate ?

> I think you should have a boolean logrotate attribute for setting logrotate on or off.
:+1: 

Could we add documentation as well? And tests?

File: libraries/varnish_log.rb [hunk 1]
```diff
@@ -102,6 +103,20 @@ def configure_varnish_log
           notifies :restart, "service[#{new_resource.log_format}]", :delayed
         end
 
+        template "#{new_resource.logrotate_path}/#{new_resource.log_format}" do
+          source 'lib_logrotate_varnishlog.erb'
+          path "#{new_resource.logrotate_path}/#{new_resource.log_format}"
```
Comment (jujugrrr) [inline libraries/varnish_log.rb]: This line might generate a file in `/etc/logrotate.d/varnishlog` or  `/etc/logrotate.d/varnishncsa`. Will it override the default Varnish logrorate provided by the package?

It seems on Ubuntu by example the default file is `/etc/logrotate.d/varnish`

Comment (prometheanfire) [inline libraries/varnish_log.rb]: Do we care about the package's file?

**Comment by prometheanfire** [comment]
Ya, I'm kinda torn about this, on one hand we shouldn't be combining functions in a single cookbook like this I think.  On the other this is all about varnish, and we don't make it depend on another cookbook, so I'm kinda torn.  I will be adding a switch or making creation of the log file a separate item though.  So it can be 'enabled' separate of the nsca/log functionality.

**Comment by prometheanfire** [comment]
actually, what do you think of just having the logrotate file available but not installing it automatically?  People can still use it if they want and will not be force to use it.  It also rids us of the problem of combining too many things into the library.

**Comment by mattjbarlow** [comment]
I think the best option is just to have a toggle on / off for logrotate and drop the file in if people want it. That way, if someone wants to do their own thing with logrotate then they can simply turn it off. Personally, I'd be likely to enable it in my cookbooks so that I don't have to bother with setting up logrotate separately.

I also don't foresee any conflicts with the package logrotate. At worst, you will have an extra logrotate file that is not rotating anything.

**Comment by prometheanfire** [comment]
we'll be ready to merge after this last commit passes (it's just a readme update)

**Comment by jujugrrr** [comment]
> I also don't foresee any conflicts with the package logrotate. At worst, you will have an extra logrotate file that is not rotating anything.

It sounds dirty, I'd rather override the default one, the toggle is to manage logrotate with Varnish cookbook, not to add some other configuration for logrotate that might conflict.

Can we write a test for the switch?

**Comment by prometheanfire** [comment]
@mattjbarlow commented that it rotates another log file, so we are fine, no conflict.

**Comment by jujugrrr** [comment]
> @mattjbarlow commented that it rotates another log file, so we are fine, no conflict.

Great, then we just need some tests :-)

**Comment by mattjbarlow** [comment]
Tests were added and are passing. We decided not to rewrite the existing recipes in this release, because of some minor differences and the removal of Ubuntu 12.04 support. We will rewrite the recipes in the next release, and in the release after that they will be removed.

**Comment by shmuel-krakower** [comment]
It is somewhat misleading that there's still the node['varnish']['log_daemon'] while there's a new varnish_log resource. At least for me.
What should I do if I want to have the log OFF and the ncsalog ON ?

**Comment by prometheanfire** [comment]
in the future you'll call the varnish_log resource directly (and more dynamically).  You can see our fixture cookbook as an example.

https://github.com/rackspace-cookbooks/varnish/blob/master/test/fixtures/cookbooks/install_varnish/recipes/distro_install.rb#L49-L63

**Comment by mattjbarlow** [comment]
If you are using the recipes today, you can set the `node['varnish']['log_daemon']` attribute to false, and then call the varnish_log resource with `log_format` value set to `'varnishncsa'`

The node attributes will be phased out in a future release since they are confusing, as you pointed out. We wanted to give people time to get used to the new resources before removing the node attributes.

**Pull request merged**